style.go

```diff
@@ -297,6 +297,26 @@ func (s *Style) synthesisable(ttype TokenType) bool {
 	return ttype == LineHighlight || ttype == LineNumbers || ttype == LineNumbersTable
 }
 
+// ClampBrightness returns a new derived Style with each entry's Colour brightness
+// clamped within [min, max] in the range [0, 1]. See Colour's ClampBrightness function.
+func (s *Style) ClampBrightness(min, max float64) *Style {
```

Comment:
The reason I asked for this on `StyleBuilder` is because that is the established pattern for creating a new style from an existing one. Having this method on `Style` introduces a secondary code path for that pattern, which I do not want.

Reply:
Sure thing, I've changed this to a `Transform` function on `StyleBuilder`. It has deduping code that's very similar to that in `Style.Types`, but that seems unavoidable.